### PR TITLE
feat: publish aztec build of noir_wasm

### PR DESCRIPTION
**Pull request by alexghr**
# Description

~~This PR adds a new build target: `noir_wasm_aztec`.~~
Modifies publishing script to consider `npm-tag` input to a workflow, based on which will enable feature flag to compile `aztec` specific library version.

## Problem\*

We'll need a noir_wasm build with the `aztec` feature enabled in the frontend for AztecProtocol/aztec-packages#2644.

## Summary\*

~~The new build target is the same as `noir_wasm` but enables the `noirc_frontend/aztec` feature at build time.~~
Builds aztec version with feature flag by adding `default = ["aztec"]` to `noirc_frontend/Cargo.toml`.
The build artifact will have a pre-release version of `x.y.z-<short-rev>-aztec` and will be published to npm under the `aztec` tag.

## Documentation

N/A

## Additional Context

N/A

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by TomAFrench** [comment]
We've changed the publishing workflows in #3029, can you update this PR to match this?

**Comment by kobyhallx** [comment]
Made an alternative proposal in 
4dcde8f: chore: simpler aztec publishing

**Review by alexghr — COMMENTED** [review]
Looks great, thanks @kobyhallx!

File: .github/workflows/publish-es-packages.yml [hunk 1]
```diff
@@ -72,11 +93,26 @@ jobs:
...
 
+      - name: Prepare nightly version
+        if: ${{ inputs.npm-tag == 'nightly' }}
+        run: |
+          sudo apt-get install jq
+          yarn nightly:version
+
+      - name: Update Version as specific
+        run: |
+          jq '.version = .version + "-${{ inputs.npm-tag }}"' package.json > package-tmp.json && mv package-tmp.json package.json
```
[the remark this answers is not in the document]
Comment (kobyhallx) [reply .github/workflows/publish-es-packages.yml:112]: yes, good catch

**Review by TomAFrench — APPROVED** [review]
LGTM

**Pull request merged**